server/aesdsocket.c

```diff
@@ -0,0 +1,202 @@
+    int bytes_from_client;
+    int buf_size_count = 0;
+    int bytes_pending = 0;
+    int fsize = 0;
+    char addrstr[INET6_ADDRSTRLEN];
+    int newlineflag = 0;
+
+    // Socket Variables & Declaration
+    int new_conn;
+    int status;
+    struct addrinfo hints;
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_INET;
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;
+    struct sockaddr_storage test_addr;
+    socklen_t addr_size;
+
+    // -------------------------GetAddrInfo-Socket-Bind-----------------------------------------------------------------------------------
+    if ((status = getaddrinfo(NULL, "9000", &hints, &servinfo)) != 0)
+    {
+        fprintf(stderr, "getaddrinfo error: %s\n", gai_strerror(status));
+        exit(1);
+    }
+
+    sockfd = socket(servinfo->ai_family, servinfo->ai_socktype, servinfo->ai_protocol);
+    if (sockfd == -1)
+        return -1;
+
+    int yes = 1;
+    if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof yes) == -1)
+    {
+        perror("setsockopt");
+        exit(1);
+    }
+
+    if (bind(sockfd, servinfo->ai_addr, servinfo->ai_addrlen) == -1)
+        return -1;
+    // -------------------------------Create-Daemon--after-bind-------------------------------------------------------------------------
+    if (argc > 1)
+    {
+        char *daemon = "-d";
+        if (strcmp(argv[argc - 1], daemon) == 0)
+        {
+            pid_t pid;
+            pid = fork();
+
+            if (pid == -1)
+                return -1;
+            else if (pid != 0)
+                exit(EXIT_SUCCESS);
+
+            if (setsid() == -1)
+                return -1;
+
+            if (chdir("/") == -1)
+                return -1;
+
+            open("/dev/null", O_RDWR);
+            dup(0);
+            dup(0);
+        }
+    }
+
+    // -------------------------------Listen--------------------------------------------------------------------------------------------
+    if (listen(sockfd, 5) == -1)
+        return -1;
+
+    fd = open(file, O_RDWR | O_CREAT | O_TRUNC, S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+
+    // In case of an error opening the file
+    if (fd == -1)
+    {
+        syslog(LOG_ERR, "Error opening file with error: %d", errno);
+    }
+
+    while (1)
+    {
+        addr_size = sizeof test_addr;
+        new_conn = accept(sockfd, (struct sockaddr *)&test_addr, &addr_size);
+
+        struct sockaddr_in *p = (struct sockaddr_in *)&test_addr;
+        syslog(LOG_DEBUG, "Accepted connection from %s", inet_ntop(AF_INET, &p->sin_addr, addrstr, sizeof(addrstr)));
+
+        while (1)
+        {
+            bytes_from_client = recv(new_conn, buffer, sizeof(buffer), 0);
+
+            // When no bytes are sent, connection is closed
+            if (bytes_from_client == 0)
+                break;
+
+            // Check if the received packet ends with newline
+            if (buffer[bytes_from_client - 1] == '\n')
```

Comment:
@neohere97 what assumption are you making about the number of packets in a `recv` buffer here?  What if two packets were in one `recv` buffer?  What if one packet and a part of a second packet were in one `recv` buffer?

Reply:
Thanks for the code review, Dan. I am working to fix the design issues in this code while I work on Assignment 6. 
Was in a bit of a rush with Assignment 5 and ended up making a lot of design compromises. Like you mentioned during the class a more robust test script would have failed my implementation.